crates/autopilot/src/solvable_orders.rs

```diff
@@ -604,19 +611,26 @@ async fn update_task(
     }
 }
 
-fn get_orders_with_native_prices(
+async fn get_orders_with_native_prices(
     orders: Vec<Order>,
     native_price_estimator: &CachingNativePriceEstimator,
     metrics: &Metrics,
+    additional_tokens: &[H160],
+    timeout: Duration,
 ) -> (Vec<Order>, BTreeMap<H160, U256>) {
-    let traded_tokens = orders
+    let mut traded_tokens = orders
         .iter()
         .flat_map(|order| [order.data.sell_token, order.data.buy_token])
-        .collect::<HashSet<_>>()
-        .into_iter()
-        .collect::<Vec<_>>();
+        .collect::<HashSet<_>>();
 
-    let prices = get_native_prices(&traded_tokens, native_price_estimator);
+    traded_tokens.extend(additional_tokens);
```

Comment:
You can probably `.chain(additional_tokens)` when building `traded_tokens` to not make it `mut`.